aviatrix/resource_vpc.go

```
@@ -0,0 +1,188 @@
+				Required:    true,
+				Description: "Region where this gateway will be launched.",
+			},
+			"vpc_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the VPC to be created.",
+			},
+			"vpc_cidr": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Subnet of the VPC to be created.",
+			},
+			"aviatrix_transit_vpc": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specify the VPC as Aviatrix Transit VPC or not.",
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the VPC to be created.",
+			},
+			"subnets": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Subnets of the VPC to be created.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cidr": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Subnet cidr.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Subnet name.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVpcCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+	vpc := &goaviatrix.Vpc{
+		CloudType:   d.Get("cloud_type").(int),
+		AccountName: d.Get("account_name").(string),
+		VpcRegion:   d.Get("vpc_region").(string),
+		VpcName:     d.Get("vpc_name").(string),
+		VpcCidr:     d.Get("vpc_cidr").(string),
+	}
+
+	aviatrixTransitVpc := d.Get("aviatrix_transit_vpc").(bool)
+	if aviatrixTransitVpc {
+		vpc.AviatrixTransitVpc = "yes"
+	} else {
+		vpc.AviatrixTransitVpc = "no"
+	}
+
+	if vpc.AviatrixTransitVpc == "yes" {
+		log.Printf("[INFO] Creating a new Aviatrix Transit VPC: %#v", vpc)
+	} else {
+		log.Printf("[INFO] Creating a new VPC: %#v", vpc)
+	}
+
+	err := client.CreateVpc(vpc)
+	if err != nil {
```

Should you ignore error if same VPC name is already created?

---

Yes. Done.